packages/clay-management-toolbar/src/ActiveState.tsx

```diff
@@ -0,0 +1,248 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLink from '@clayui/link';
+import Item from './Item';
+import ItemList from './ItemList';
+import React from 'react';
+import {ClayDropDownWithBasicItems} from '@clayui/drop-down';
+import {IItem} from '@clayui/drop-down/src/DropDownWithBasicItems';
+import {IPropsBase} from './types';
+
+interface IActionItems extends IItem {
+	quickAction?: boolean;
+}
+
+interface IProps extends IPropsBase {
+	actionItems?: Array<IActionItems>;
+	clearSelectionURL?: string;
+	onClearSelectionClick?: (
+		e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+	) => void;
+	onQuickActionClick?: (
+		e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+	) => void;
+	onSelectAllClick?: (
+		e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+	) => void;
+	selectAllURL?: string;
+	selectedItems?: number;
+	showSelectAllButton?: boolean;
+	supportsBulkActions?: boolean;
+	totalItems?: number;
+}
+
+const ActiveState: React.FunctionComponent<IProps> = ({
+	actionItems = [],
+	checkboxChecked = false,
+	clearSelectionURL,
+	disabled = false,
+	filterItems,
+	labels = {
+		checkbox: 'Select Items',
+		clearSelection: 'Clear',
+		selectAll: 'Select all',
+	},
+	onCheckboxChange = () => {},
+	onClearSelectionClick = () => {},
+	onInfoButtonClick = () => {},
+	onQuickActionClick = () => {},
+	onSelectAllClick = () => {},
```

Comment:
All of these default no-op functions seem strange. Do we intend for people to use this component without providing these callbacks? It seems like there would be missing functionality if they don't provide a proper callback.

Reply:
Yes this is really weird, this is a very large component with many use cases and n features so it's possible that you just render a Search for example without needing all the other components just defining it as the management toolbar will be rendered through the API.

Reply:
If thats the case, we should still likely still keep these functions as undefined, and then wrap its uses with an if check to make sure its defined. Otherwise, if we have `onInfoButtonClick` defaulted to a no-op, we will show that button but have no functionality for it.

Reply:
Hm makes sense. I will take care of adding them to conditional on their features.